pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -243,6 +250,34 @@ func (c *Controller) Delete(typ, name, namespace string) error {
 	return errUnsupported
 }
 
+func (c *Controller) sync(collection string) {
+	c.syncedMu.Lock()
+	c.synced[collection] = true
+	c.syncedMu.Unlock()
+}
+
+func (c *Controller) incrementalUpdate(innerStore, prevStore map[string]map[string]*model.Config) error {
+	for ns, byName := range innerStore {
+		for name, config := range byName {
+			if prevByNamespace, ok := prevStore[ns]; ok {
+				if prevConfig, ok := prevByNamespace[name]; ok {
+					if config.ResourceVersion != prevConfig.ResourceVersion {
+						if se, ok := config.Spec.(*networking.ServiceEntry); ok {
+							endpoints := serviceEndpoints(se, name, ns)
+							hostname := fmt.Sprintf("%s.%s.svc.%s", name, ns, c.options.DomainSuffix)
```

Comment:
nit: Don't you Sprintf for hot paths ... it's slow and creates garbage.  Use regular string concat, it's optimized:

```golang
hostname := name + "." + namespace + ".svc." + c.options.DomainSuffix
```